### PR TITLE
Controller instance integration tests

### PR DESCRIPTION
**Pull request by staebler**
Add integration tests that tests scenarios around creating and updating ServiceInstances.

### CODE REVIEW

File: test/integration/controller_instance_test.go [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+	// avoid error `servicecatalog/v1beta1 is not enabled`
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/test/util"
+)
+
+// TestCreateServiceInstanceNonExistentClusterServiceClassOrPlan tests that a ServiceInstance gets
+// a Failed condition when the service class or service plan it references does not exist.
+func TestCreateServiceInstanceNonExistentClusterServiceClassOrPlan(t *testing.T) {
```
Comment (n3wscott) [inline test/integration/controller_instance_test.go]: Should there also be a test for setting both external and k8s names? Perhaps a test if they don't agree?

Comment (staebler) [reply test/integration/controller_instance_test.go]: In short, yes, there should be a test. I would argue that this is not the best place for the test. The tests in these files are intended to test reconciliation in the controller. The API Server will reject a ServiceInstance that has both the external and k8s names set.

**Comment by pmorie** [comment]
@staebler @jpeeler should this have do-not-merge?

**Comment by pmorie** [comment]
@staebler @jpeeler Okay - I'm not clear on whether this PR is meant to be mergeable as-is, or whether jpeeler is going to open a new PR for this stuff.  Can you clarify?

**Comment by staebler** [comment]
@pmorie My understanding is that this PR is meant to merge as is. @jpeeler will be creating additional tests that will come in future PRs.

**Comment by jpeeler** [comment]
Yeah I just pulled this PR and kept on going. So once this merges I'll just rebase.

**Comment by kibbles-n-bytes** [comment]
Fantastic work with this. LGTM.

**Pull request merged**